### ISSUE TITLE
"but no match was found" is shown in explanation

### ISSUE THREAD

**Issue by robstoll** [robstoll/atrium#1364]
### Affected Version

0.18.0

### API

fluent-en_GB, infix-en_GB

### Platform

jvm, js

### Kotlin Version

1.4

### How to reproduce the problem?

```kotlin
expect(listOf("hello", "world", "foo", "bar")).notToContain.entry {
            toContain("hello")
        }
```

### Describe the bug

output
```
I expected subject: [hello, world, foo, bar]        (java.util.Arrays.ArrayList <1643453315>)
◆ not to contain: 
  ⚬ an element which needs: 
      » to contain: 
        ⚬ value: "hello"        <2088540432>
            » but no match was found                         <------------ this should not be shown here
      ❗❗ following elements were mismatched: 
         ⚬ index 0: "hello"        <2088540432>
```

### Expected behaviour

output
```
I expected subject: [hello, world, foo, bar]        (java.util.Arrays.ArrayList <1643453315>)
◆ not to contain: 
  ⚬ an element which needs: 
      » to contain: 
        ⚬ value: "hello"        <2088540432>
      ❗❗ following elements were mismatched: 
         ⚬ index 0: "hello"        <2088540432>
```

**Comment by WesleyJammer** [comment]
@robstoll i am motivated to take on this one, if that is possible!

**Comment by robstoll** [comment]
Well well... I just realised that this particular text will be dropped entirely with the new reporting. But a similar problem still remains and still needs to be fixed (I think, might be I already fixed it). You need to base your work on the feature/proof-based-report branch. Start off with creating a test in CreateReportTest (see existing proofExplanation_... tests). Following the things we want to achieve:
- UsageHintGroup should **not** be shown if within a ProofExplanation
- DebugGroup should **not** be shown if within a ProofExplanation
- InformationGroup should be shown within a ProofExplanation
- FailureExplanationGroup should be shown within a ProofExplanation
- also address nested cases, e.g. ProofExplanation -> FailureExplanationGroup -> ProofExplanation -> UsageHintGroup 

Let me know if you need more Information

**Comment by WesleyJammer** [comment]
@robstoll So if i am understanding correctly the original issue is fixed, but a similar issue is occuring with the above parts in ProofExplanation?

**Comment by robstoll** [comment]
@WesleyJammer yes, UsageHintGroup and DebugGroup are most likely shown if they occur in a ProofExplanation currently, but should not

**Comment by WesleyJammer** [comment]
@robstoll Thanks for the information! i will look into the issue and when i encounter problems or if i have a question i will contact you.

**Comment by robstoll** [comment]
Sorry for the delay, that's now the third time I see your issue and start writing an answer. Let's hope I am not getting interrupted again 😁Usually repos welcoming contributors have a CONTRIBUTING.md in the .github folder. Here the one of atrium: https://github.com/robstoll/atrium/blob/main/.github/CONTRIBUTING.md

I include it in good first issues, following the section:

## Your first contribution?
- Write a comment `I'll work on this` if you would like to take this issue over. 
  This way we get the chance to revise the description in case things have changed in the meantime, we might give you additional hints and we can assign the task to you, so that others do not start as well.
- See [Your first code contribution](https://github.com/robstoll/atrium/blob/main/.github/CONTRIBUTING.md#your-first-code-contribution) for guidelines.  
- Do not hesitate to ask questions here or to contact us via [Atrium's slack channel](https://kotlinlang.slack.com/team/U3DE1TXKP) if you need help
  ([Invite yourself](https://slack.kotlinlang.org/) in case you do not have an account yet).

**Comment by WesleyJammer** [comment]
@robstoll haha no problem at all. I am already more than happy that you help me with my questions! i will look into the guidelines!

**Comment by WesleyJammer** [comment]
@robstoll when building the main project i get a build failed message because 19 tests are failed, including one that says that i dont have one of the required permissions and it is about creating cymbolic links.

**Comment by robstoll** [comment]
failed or ignored? Are you on windows?

**Comment by WesleyJammer** [comment]
first ignored, but the build ends with 19 failed tests and then the message build failed. The test results you can find in build - reports- test- jvmTest - index.html. yes i am on windows.

**Comment by robstoll** [comment]
The CI is green for windows but a lot of tests are ignored since there are features which Windows does not support:
https://github.com/robstoll/atrium/actions/runs/13435355382/job/37536311080#step:5:510

I guess you encounter a permission issue. Are you able to create symlinks (`mklink link_target actual_file`) without admin rights? Anyway... you can ignore those failures for know. I'll see If I am able to reproduce your error but I first need to setup a windows VM. We can do a quick pair programming to try to resolve your issue if you want.

**Comment by robstoll** [comment]
@WesleyJammer are you running inside WSL by now? if so you can execute the following:
```
sudo apt-get install -y openjdk-11-jdk openjdk-11-source
```

**Comment by robstoll** [comment]
Not at all, take your time and make sure you pull upstream so that when you start/continue your work, it is based on latest.

**Comment by robstoll** [comment]
rebase on feature/proof-based-report. Not sure when I last worked on it, so could be you are already up-to-date

**Comment by WesleyJammer** [comment]
@robstoll about creating a test. i am creating a simple try out test and have a question?


```java
@Test
 fun proofExplanation_hidesUsageHintAndDebug_showsInformationAndFailureExplanation()
    {
        val builder = buildRootGroup{
            proofGroup(Text("not to contain"), Text.EMPTY)
            {
                invisibleFailingProofGroup {

                    proofExplanation {
                        simpleProof(Text("to be greater than"), 2) { false }
                        usageHintGroup {
                            add(ErrorMessages.FORGOT_DO_DEFINE_EXPECTATION)
                            add(ErrorMessages.DEFAULT_HINT_AT_LEAST_ONE_EXPECTATION_DEFINED)
                        }
                    }
                }
            }


        }

        expectForReporterWithoutAnsi(
            builder,
            """
            |a verb : "representation"
            |(f) not to contain       :${' '}
            |    » to be greater than : 2
      ()`
            """.trimMargin()
        )
    }
```
am i on the right way to check if a usageHintGroup is not visible when inside a proofExplanation or do i need to find a way to compare this answer to the outcome of the actual reporter?

**Comment by WesleyJammer** [comment]
@robstoll okey great! Then the test i showed you confirms that at this moment the ``` proofExplanation ``` expects a ``` usageHintGroup ``` to be visible because my test fails. The test expects ```  |      (u) You forgot to create expectations in the expectationCreator-lambda
            |      (u) Sometimes you can use an alternative to `{ }` For instance, instead of `toThrow<..> { }` you should use `toThrow<..>()` ``` to be visible in the ``` expectForReportWithoutAnsi ``` . 

Is it fine for you that at first i only focus on fixing the ``` usageHintGroup ``` problem, before i start to look at the ``` debugGroup ``` etc.?

**Comment by WesleyJammer** [comment]
@robstoll i think i fixed the first part of excluding UsageHintGroup from the reporter output, when inside a ProofExplanation. I made a pull request to merch my changes into the proof-based-report branch. The only thing is i have merch conflicts that can't be resolved in the editor somehow, so i hope you can help me out with this. I also hope the change i made is the result you want and i am looking forward to your answer.
